tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs

```diff
@@ -566,16 +561,22 @@ public void RunWithInterpreterEnabled ([Values (false, true)] bool isRelease)
 
 			ClearAdbLogcat ();
 			RunAdbCommand ("shell setprop debug.mono.log all");
+			var logProp = RunAdbCommand ("shell getprop debug.mono.log")?.Trim ();
+			Assert.AreEqual (logProp, "all", "The debug.mono.log prop was not set correctly.");
+
 			if (CommercialBuildAvailable)
 				Assert.True (builder.RunTarget (proj, "_Run"), "Project should have run.");
 			else
 				AdbStartActivity ($"{proj.PackageName}/{proj.JavaPackageName}.MainActivity");
 
-			var logcatFilePath = Path.Combine (Root, builder.ProjectDirectory, "logcat.log");
-			var didStart = WaitForActivityToStart (proj.PackageName, "MainActivity", logcatFilePath, 30);
-			RunAdbCommand ("shell setprop debug.mono.log \"\"");
+			var didStart = WaitForActivityToStart (proj.PackageName, "MainActivity", "startup-logcat.log", 120);
+			RunAdbCommand ("shell setprop debug.mono.log \"''\"");
+			logProp = RunAdbCommand ("shell getprop debug.mono.log")?.Trim ();
+			Assert.AreEqual (logProp, string.Empty, "The debug.mono.log prop was not unset correctly.");
 			Assert.True (didStart, "Activity should have started.");
-			Assert.IsTrue (File.ReadAllText (logcatFilePath).Contains ("Enabling Mono Interpreter"), "logcat output did not contain 'Enabling Mono Interpreter'.");
+			Assert.True (MonitorAdbLogcat ((line) => {
+				return line.Contains ("Enabling Mono Interpreter");
+			}, Path.Combine (Root, builder.ProjectDirectory, "interpreter-logcat.log"), 120), "logcat output did not contain 'Enabling Mono Interpreter'.");
```

Comment:
While I realize this "pattern" is used elsewhere in this file, *I* found this line confusing to read, as the `120)` appears to be "free floating", and I no longer see what it is applied to.

Thus, could you split things up for clarity?

```csharp
Func<string, bool> checkForInterpeter = line => {
    return line.Contains ("Enabling Mono Interpreter");
};
Assert.IsTrue (
    condition: MonitorAdbLogcat (
        action: checkForInterpreter,
        logcatFilePath: Path.Combine (Root, builder.ProjectDirectory, "interpreter-logcat.log"),
        timeout: 120),
    message: "logcat output did not contain 'Enabling Mono Interpreter'.");
```